dg/commands/cpd4/uninstall.py

```diff
@@ -0,0 +1,73 @@
+#  Copyright 2021 IBM Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Optional
+
+import click
+
+import dg.config
+import dg.config.cluster_credentials_manager
+import dg.lib.click.utils
+import dg.lib.cluster
+import dg.utils.download
+import dg.utils.network
+
+from dg.lib.cloud_pak_for_data.cpd_4_0_0.cpd_manager import (
+    CloudPakForDataManager,
+)
+from dg.utils.logging import loglevel_command
+
+
+@loglevel_command(
+    context_settings=dg.lib.click.utils.create_default_map_from_dict(
+        dg.config.cluster_credentials_manager.cluster_credentials_manager.get_current_credentials()
+    )
+)
+@click.option("--server", help="OpenShift server URL", required=True)
+@click.option("--username", help="OpenShift username")
+@click.option("--password", help="OpenShift password")
+@click.option("--token", help="OpenShift OAuth access token")
+@click.option("--delete-project", help="Delete project", is_flag=True)
+@click.option("--force", "-f", help="Skip confirmation", is_flag=True)
+@click.option("--project", default="zen", help="Project where the Cloud Pak for Data control plane is installed")
+@click.pass_context
+def uninstall(
+    ctx: click.Context,
+    server: str,
+    username: Optional[str],
+    password: Optional[str],
+    token: Optional[str],
+    delete_project: bool,
+    force: bool,
+    project: str,
+):
+    """Uninstall IBM Cloud Pak for Data"""
+
+    if not force:
+        click.confirm(
+            f"Do you really want to uninstall IBM Cloud Pak for Data on server {server} from project '{project}'?",
+            abort=True,
+        )
+
+        click.confirm(
+            "Have you uninstalled any services that are installed in the same project (or tethered projects) as the "
```

Comment:
Can you please elaborate what this means? I'm probably missing context, but at the moment I don't know what this prompt means and what the consequences are.

Reply:
This question corresponds to [this section](https://www.ibm.com/docs/en/cloud-paks/cp-data/4.0?topic=installing-uninstalling-cloud-pak-data#rhos-uninstall__dependencies) in the official IBM Cloud Pak for Data documentation:

> Ensure that you uninstall any services that are installed in the same project (or tethered projects) as the control plane before you uninstall the control plane.

Ideally, the CLI would know how to uninstall all instances of a service, the service itself, and the operator. To this end, we would have to statically store or dynamically determine the kind of the respective service instance (Db2 Data Gate: `DatagateServiceInstance`) and the service itself (Db2 Data Gate: `DatagateService`). According to the documentation, each operator should be uninstalled using the IBM Cloud Pak CLI (example: [Db2 Data Gate](https://www.ibm.com/docs/en/cloud-paks/cp-data/4.0?topic=gate-uninstalling-db2-data#operator-uninstall__uninstall-operator)), which is very unfortunate. Therefore, I would like to postpone this slightly complex task for now. However, feel free to suggest a better question.

Reply:
I agree. Let's postpone this. It's definitely not necessary for a first release. Furthermore, there is always hope that the process of uninstalling a service becomes more streamlined in CPD or Openshift.